contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -697,23 +716,23 @@ impl InterchainTokenService {
         token_metadata: TokenMetadata,
         minter: Option<Address>,
     ) -> Result<Address, ContractError> {
-        Self::ensure_token_not_registered(env, token_id.clone())?;
+        let unregistered_token_id = Self::ensure_token_not_registered(env, token_id)?;
 
         let token_address = deployer::deploy_interchain_token(
             env,
             Self::interchain_token_wasm_hash(env),
             minter,
-            token_id.clone(),
+            unregistered_token_id.clone().into(),
```

Comment:
```suggestion
            unregistered_token_id.clone(),
```
why not use it for this method too to ensure safety?

Reply:
updated